source-mysql/main.go

```diff
@@ -244,11 +251,38 @@ func (db *mysqlDatabase) connect(ctx context.Context) error {
 		return fmt.Errorf("unable to connect to database: %w", err)
 	}
 
-	// Infer the location in which captured DATETIME values will be interpreted.
-	if loc, err := queryTimeZone(conn); err == nil {
+	// Infer the location in which captured DATETIME values will be interpreted. The system variable
+	// 'time_zone' is preferentially used if it is set on the database.
```

Comment:
It seems like it might be better to prioritize the config property over the database setting?

I don't have a really strong opinion here, but usually for this sort of "leave blank to autodetect" setting it feels very surprising for explicitly setting the value to _not_ take precedence over the autodetection logic.

Reply:
Yeah I think you're right, stepping back a bit. I was approaching this kind of reluctantly, leaving it as a "last resort", but it makes more sense for an explicit configuration to take precedence if set rather than for it to not take effect under situations the user might not be aware of.